flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/PointwiseBlockingResultInfo.java

```diff
@@ -60,4 +61,23 @@ public long getNumBytesProduced() {
                 .flatMapToLong(Arrays::stream)
                 .reduce(0L, Long::sum);
     }
+
+    @Override
+    public long getExecutionVertexInputNumBytes(
+            IndexRange partitionIndexRange, IndexRange subpartitionIndexRange) {
+        checkState(
+                subpartitionBytesByPartitionIndex.size() > partitionIndexRange.getEndIndex(),
+                "Not all partition infos are ready");
```

Comment:
I think we should check the `subpartitionBytesByPartitionIndex.get(i)` is not null, and the check the `subpartitionIndexRange.getEndIndex() < subpartitionBytesByPartitionIndex.get(i).lengh`

Reply:
Thanks for reminding, I have added the check.